packages/algolia/src/client.ts

```diff
@@ -0,0 +1,58 @@
+import { SearchIndex } from 'algoliasearch';
+import { SearchOptions, SearchResponse } from '@algolia/client-search';
+import { ResearchOutputResponse, UserResponse } from '@asap-hub/model';
+
+export type EntityResponses = {
+  'research-output': ResearchOutputResponse;
+  user: UserResponse;
+};
+
+export type EntityRecord<T extends keyof EntityResponses> =
+  EntityResponses[T] & {
+    objectID: string;
+    __meta: {
+      type: T;
+    };
+  };
+
+export const getEntityType = (
+  entity: EntityResponses[keyof EntityResponses],
+): keyof EntityResponses => {
+  if ('title' in entity) {
```

Comment:
This feels like a weak property. Maybe we add a title to user at some point...

Reply:
hmmm.. i could use more properties to distinguish but I think there is not much else to tell them apart, but yeah agreed - just the title isn't going to cut it